src/Stratis.Bitcoin/P2P/ISelfEndpointTracker.cs

```diff
@@ -7,6 +11,15 @@ namespace Stratis.Bitcoin.P2P
     /// </summary>
     public interface ISelfEndpointTracker
     {
+        /// <summary>Update external IP address and peer score of the node.</summary>
+        void UpdateAndAssignMyExternalAddress(IPEndPoint ipEndPoint, int? ipEndPointPeerScore);
```

Comment:
We don't need the second argument nullable. It would be better if it has a default value of one. Also please add comments for parameters